lib/logflare/cluster/postgres_strategy.ex

```diff
@@ -0,0 +1,117 @@
+defmodule Logflare.Cluster.PostgresStrategy do
+  @moduledoc """
+  A libcluster strategy that uses Postgres LISTEN/NOTIFY to determine the cluster topology.
+
+  This strategy operates by having all nodes in the cluster listen for and send notifications to a shared Postgres channel.
+
+  When a node comes online, it begins to broadcast its name in a "heartbeat" message to the channel. All other nodes that receive this message attempt to connect to it.
+
+  This strategy does not check connectivity between nodes and does not disconnect them
+
+  ## Options
+
+  * `heartbeat_interval` - The interval at which to send heartbeat messages in milliseconds (optional; default: 5_000)
+
+  The magic cookie is used as the channel name.
```

Comment:
Did we check constraints around this from Postgres?

Reply:
In Supavisor and Realtime this is passed in with the config.

If we want to use the magic cookie we should probably move this and do it there.

Reply:
from my manual testing, pg doesn't accept a channel name with `:` prefix, hence why i converted it to use `Atom.to_string/1` to get the stringified atom. `Node.get_cookie/0` seems to return the cookie as an atom.
I can make this more foolproof though with some alphanumeric conversions

Reply:
Roger.  Maybe default behavior (for the eventual library) is to use the cleaned magic cookie and override (or additionally) subscribe to the channel(s) passed in the config.

Reply:
And yeah just double check because the cookie on dev could not be as long or complex as prod.